package.json

```diff
@@ -6,6 +6,8 @@
     "dev": "next dev",
     "build": "next build",
     "start": "next start",
+    "start:with-cron-jobs": "run-p start cron",
+    "cron": "tsx cron-jobs/delete_expired_ip_rules.ts",
```

Comment:
Do we really need to run this as completely separate task? It makes it more complex to run it locally and to deploy it.

Could we just run it on the next.js server side?

Reply:
I tried but didn't find any obvious way of doing it. It's easy when hosting on Vercel, but we are currently using Digital Ocean. Maybe an extra argument for migrating to Vercel one day. Digital Ocean has no built-in cron functionality for its App platform (or I wasn't able to find it, let me know if I missed something). 

We considered making an endpoint and calling it from a scheduled GitHub action with some secret, but this seemed like a simpler alternative to that.

Reply:
How about instead of having a cron task, just checking and deleting expired ip rules whenever the request to list them is sent? 
The operation can run in async manner, but just be triggered whenever the endpoint is called.

Reply:
Hmm it could work! :) Potential issues:

- If there is a lull in traffic the TTL is exceeded until someone opens the page. Not sure how likely that is but it could happen. 
- In terms of customer education, maybe a cron job is more intuitive? 

@ilfa what do you think?

Reply:
> In terms of customer education, maybe a cron job is more intuitive?

We can recommend this practice in the code comments 🙂

Reply:
Do customers really need to clean blocked IPs by cron or other algorithm?
Because I'm ok with any solution on our side.

Reply:
That was one of the pieces of feedback to Martin's original demo, that users would want to have some TTL on the rules and Cludflare does not have that option. Providing a general example of how best to achieve that functionality with Cloudflare seems valuable to me, that's why I lean towards the cron job even is it's slightly more complex. It more cleanly answers a question that is likely to come up from users.